frontend/packages/ceph-storage-plugin/src/components/create-storage-system/external-storage/types.ts

```diff
@@ -0,0 +1,95 @@
+import { K8sResourceKind } from '@console/internal/module/k8s';
+import { IP_FAMILY } from '../../../constants';
+
+/**
+ *  Configures a new external storage vendor to the Storage System Creation wizard.
+ */
+export type ExternalStorage = {
+  /** Display name of the external storage vendor. */
+  displayName: string;
+
+  /** The model referring the `apiGroup`,`apiVersion` and `kind` of the external storage vendor's CRD. */
+  model: ExtensionK8sModel;
+
+  /** A React Functional Component to input the connection details of the external storage vendor. */
+  Component: React.FunctionComponent<ExternalComponentProps<{}>>;
+
+  /**  Handler function to create external storage storage vendor CR or resources. */
+  createPayload: CreatePayload<{}>;
+
+  /**  Handler function to validate the storage class page in order to move to the next step of wizard */
+  canGoToNextStep: CanGoToNextStep<{}>;
+};
+
+/** Props for `ExternalStorage.Component` to input the connection details of the external storage vendor. */
+export type ExternalComponentProps<S extends ExternalState> = {
+  /** The state of the `ExternalStorage.Component`. */
+  formState: S;
+
+  /** The callback for setting the state of `ExternalStorage.Component` */
+  setFormState: (field: string, value: ExternalStateValues) => void;
+
+  /** The translation function for externalizing the text. */
+  t: TFunction;
+};
+
+/**
+ *  @function CreatePayload<S>
+ *
+ *    @param {string} systemName
+ *    The name of the external storage system requried for the creation of the external custom resource.
+ *
+ *    @param {S extends ExternalState} state
+ *    The other fields of the create storage class form.
+ *
+ *    @param {ExtensionK8sModel} model
+ *    The model referring the `apiGroup`,`apiVersion` and `kind` of the external storage vendor's CRD.
+ *
+ *    @param {string} storageClassName
+ *    The name of the of the storage class.
+ *
+ *    @returns {Payload} An array of payloads of `Payload` type.
+ */
+export type CreatePayload<S extends ExternalState> = (
+  systemName: string,
+  state: S,
+  model: ExtensionK8sModel,
+  storageClassName: string,
+) => Payload[];
+
+export type Payload = { model: ExtensionK8sModel; payload: K8sResourceKind };
+
+/**
+ *  @function CanGoToNextStep<S>
+ *
+ *    @param {S extends ExternalState} state
+ *    The other fields of the create storage class form.
+ *
+ *    @param {string} storageClassName
+ *    The name of the of the storage class.
+ *
+ *    @returns {boolean} A boolean value.
+ */
+export type CanGoToNextStep<S extends ExternalState> = (
+  state: S,
+  storageClassName: string,
+) => boolean;
+
+/**
+ * State for external components
+ */
+export type ExternalState = ConnectionDetailsState | {};
+
+export type ExternalStateValues = ValuesOfUnion<ExternalState>;
+
+export type ConnectionDetailsState = {
+  fileData: string;
+  isRejected: boolean;
+  isLoading: boolean;
+  fileName: string;
+  ipFamily: IP_FAMILY;
+};
```

Comment:
```suggestion
export type ConnectionDetailsState = {
  fileData: string;
  isRejected: boolean;
  isLoading: boolean;
  fileName: string;
  ipFamily: IP_FAMILY;
};
```
This is strictly related to RHCS. Why is is part of types here? Can't this be `RHCSFormState` or something that signifies that it's related to RHCS?